multi-agent/lib/mas/session/execution/foreground_runner.py

```
@@ -97,31 +110,47 @@ def _run_streaming(
+        )
+        self._binder.bind_all(session.session_registry, bindings)
 
         result: dict = {"state": None, "error": None}
 
         def _execute() -> None:
             try:
                 result["state"] = session.executable_graph.run(
-                    session.graph_state, session_id=session.get_run_id(),
+                    session.graph_state, session_id=run_id,
                 )
             except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)